packages/driver/src/cy/commands/asserting.ts

```diff
@@ -7,6 +7,30 @@ import $errUtils from '../../cypress/error_utils'
 const reExistence = /exist/
 const reHaveLength = /length/
 
+const onBeforeLog = (log, command, commandLogId) => {
+  log.set('commandLogId', commandLogId)
```

Comment:
I tried to use the log id initially, you can't set a log's ID:

```
    // dont ever allow existing id's to be mutated
    if (this.attributes.id) {
      delete this.obj.id
    }
```

because the ID is meant to be *completely unique* - to identify a log instance, across the entire test. Updating it in the log would also require updating the mapping of id -> Log instance that the `LogManager` maintains, which begins to look like the same level of complexity as having its own attribute.

Also important to note that in the current implementation, the commandLogId isn't globally unique - it's unique (and reproducible) only within the context of a single command. Different instances of `.should()` can generate the same commandLogId. This isn't a problem in the current implementation, because we only search `command.get('logs')` for a matching commandLogId.

Reply:
Basically, this just serves a different purpose than the log's id. It's a bit of state-tracking for an instance of `.should()`.